jujugui/static/gui/src/app/store/env/legacy-api.js

```diff
@@ -3249,18 +2579,43 @@ YUI.add('juju-env-legacy-api', function(Y) {
         Request: request,
         Params: {Tag: userTag}
       }, handleListModels);
+    },
+
+    /**
+      Return detailed information about Juju models available for current user.
+      Under the hood, this call leverages the ModelManager ListModels and
+      ModelInfo endpoints.
+
+      @method listModelsWithInfo
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive two arguments, the first
+        an error or null and the second an object with a "models" attribute
+        containing an array of model info, each one with the following fields:
+        - tag: the original Juju model tag;
+        - name: the model name, like "admin" or "mymodel";
+        - series: the model default series, like "trusty" or "xenial";
+        - provider: the provider type, like "lxd" or "aws";
+        - uuid: the model unique identifier;
+        - serverUuid: the corresponding controller unique identifier;
+        - ownerTag: the Juju tag of the user owning the model;
+        - life: the lifecycle status of the model: "alive", "dying" or "dead";
+        - isAlive: whether the model is alive or dying/dead;
+        - isAdmin: whether the model is an admin model;
+        - lastConnection: the date of the last connection as a string, e.g.:
+          '2015-09-24T10:08:50Z' or null if the model was never connected to;
+        - err: a message describing a specific model error, or undefined.
+      @return {undefined} Sends a message to the server only.
+    */
+    listModelsWithInfo: function(callback) {
+      // TODO frankban: implement by only using listModels.
+      callback('not implemented', null);
     }
 
   });
 
-
-  environments.endpointToName = endpointToName;
-  environments.createRelationKey = createRelationKey;
+  environments.legacyCreateRelationKey = createRelationKey;
   environments.GoLegacyEnvironment = GoLegacyEnvironment;
-  environments.lowerObjectKeys = lowerObjectKeys;
-  environments.parsePlacement = parsePlacement;
-  environments.stringifyObjectValues = stringifyObjectValues;
-  environments.machineJobs = machineJobs;
+  environments.legacyParsePlacement = parsePlacement;
 
   var KVM = {label: 'LXC', value: 'lxc'},
```

Comment:
This stuff is no longer used, I guess it can be removed, or we could restart using it in the machine view, but it's out of scope in this context.